onchaind/onchaind.c

```diff
@@ -457,6 +458,41 @@ static void ignore_output(struct tracked_output *out)
 	out->resolved->tx_type = SELF;
 }
 
+static txtypes onchain_txtype_to_wallet_txtype(enum tx_type t)
+{
+	switch (t) {
+	case FUNDING_TRANSACTION:
+		return TX_CHANNEL_FUNDING;
+	case MUTUAL_CLOSE:
+		return TX_CHANNEL_CLOSE;
+	case OUR_UNILATERAL:
+		return TX_CHANNEL_UNILATERAL;
+	case THEIR_HTLC_FULFILL_TO_US:
+	case OUR_HTLC_SUCCESS_TX:
+		return TX_CHANNEL_HTLC_SUCCESS;
```

Comment:
I had to re-read the code to see the difference between these two tx types!  `THEIR_HTLC_FULFILL_TO_US` is their unilateral close (simple case, no timeouts) and `OUR_HTLC_SUCCESS_TX` is our own unilateral close (using the sig they gave us, with timeout).

Does it make sense to distinguish these at the wallet level?  Maybe, because the second has a timeout.  Similarly OUR_HTLC_TIMEOUT_TO_US and OUR_HTLC_TIMEOUT_TX.